drivers/overlay/ov_network.go

```diff
@@ -161,6 +161,9 @@ func (n *network) destroySandbox() {
 	if sbox != nil {
 		for _, iface := range sbox.Info().Interfaces() {
 			iface.Remove()
```

Comment:
No this call is fine. Current logic is who owns the veth end, moves it back to the caller netns when its lifecycle ends.
So when the sandbox is being destroyed, sandbox code move the interface back.
Same for the other end in overlay driver, the driver moves it back.